docker/Dockerfile.optimism

```diff
@@ -0,0 +1,57 @@
+FROM alpine:3.20 AS foundry-build
+
+ARG TARGETARCH
+WORKDIR /opt
+
+RUN apk add clang lld curl build-base linux-headers git \
+    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh \
+    && chmod +x ./rustup.sh \
+    && ./rustup.sh -y
+
+RUN [[ "$TARGETARCH" = "arm64" ]] && echo "export CFLAGS=-mno-outline-atomics" >> $HOME/.profile || true
```

Comment:
So I suppose we're just trying to replicate Bulent's setup as of now, or any other reason for ARM I'm missing?

Reply:
That's a good point, TBH this line is probably useless at this point

We could probably remove it. Currently, it's simply copied from https://github.com/foundry-rs/foundry/blob/master/Dockerfile with no changes, so at least it's clear that it is replaceable by just copy pasting it without any custom changes. So I decided to keep it as is.